nativelib/src/main/scala/scala/scalanative/runtime/ieee754tostring/ryu/RyuDouble.scala

```diff
@@ -35,10 +35,10 @@ package scala.scalanative
 package runtime
 package ieee754tostring.ryu
 
-import RyuRoundingMode._
-
 object RyuDouble {
 
+  final val RESULT_STRING_MAX_LENGTH = 24
```

Comment:
Since this constant is _critical_ it would be nice to know where
it came from or was determined. Prior Ryu art?

Reply:
The name is fine, I think.   
My concern, as we have both had in other contexts, is "Where <expletives-deleted> did this magic number come from?"
I suggest a one line comment similar to 
"// Scala/Java magic number 24 is derived from original RYU C code magic number 25 (no NUL)."

Reply:
Do you have a link/reference for the RYU C code magic numbers?
Maybe I could add this to the code as well

Reply:
Found them:
https://github.com/ulfjack/ryu/blob/6f85836b6389dce334692829d818cdedb28bfa00/ryu/d2s.c#L506
https://github.com/ulfjack/ryu/blob/6f85836b6389dce334692829d818cdedb28bfa00/ryu/f2s.c#L342

Reply:
Good sleuthing!

Reply:
I took a run at chasing the origin back to the IEEE754 standard itself but could not easily find a 
baseline reference. Nothing derived in Wikipedia either.   

Ryu describes itself as "shortest length" so if they figure 24 is the longest length, that is probably
as good as an attribution as we are going to get.

We probably concur and are encouraged by:
* At least we are not leaving a magic number without attribution
* It is not something that we guessed or made up.

Sometimes that is the best we get.